cmd/tailscaled/tailscaled.go

```diff
@@ -181,6 +173,18 @@ func main() {
 		os.Exit(0)
 	}
 
+	if !strings.Contains(args.tunname, "userspace-networking") {
```

Comment:
This assumes that environments in which we userspace-networking aren't memory constrained. I'm not sure that' a good bet for things like AWS Lambda environments?

All I remeber about this tuning is that it mattered a lot to mobile devices, I don't know how we ended up making this global. If we do want to be in the business of doing this, feels like we should figure out more broadly why/when we want this aggressive setting?

Reply:
A typical outcome for this change would be 15mb -> 25mb heap. This doesn't change our churned set size, only the size of the "free list". I feel confident that for most users a 4x throughput performance and reduction in dropped traffic is worth the overhead.

This may be a problem for specific mobile environments, but I'm not sure how general of a problem it is. AIUI the latest macOS/iOS limits are 50mb now, and Tailscale's use recent Go changes reduced the cost of relocations.

Do we have platform specific budgets written down somewhere?

Reply:
We don't have strict memory budgets figured out, no. The only hard limits we have come from iOS:
 - iOS <15: 15MiB, which has to include a bunch of static program overhead (https://tailscale.com/blog/go-linker/ has the gory details, I've forgotten the nuances).
 - iOS >=15: 50MiB

AFAIK, every other platform is unconstrained on a technical level, though we're usually a system-level daemon so we should be economical where possible, in both CPU and RAM. But those aren't hard limits, just "don't be so greedy that people notice tailscale hogging resources."

Unless I've forgotten about other constraints we have, my bias would be to only set the aggressive GC tune on iOS, and leave the runtime default (or a different, less aggressive custom tune if we have one we can defend rationally) everywhere else.

cc @bradfitz, thoughts?

Reply:
I have updated the change with the following proposal:

Only apply this tuning to iOS now. I'd like to find a way to detect iOS 15 and remove it there too.